src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -2019,6 +2031,25 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::ping_storage_server() const
+  {
+    constexpr auto PING_TIMEOUT = std::chrono::milliseconds(200);
+
+    using http_client_t = epee::net_utils::http::http_simple_client;
+
+    http_client_t http_client;
+
+    const auto ip = string_tools::get_ip_string_from_int32(m_sn_public_ip);
+    const auto port = std::to_string(m_storage_port);
+
+    http_client.set_server(ip, port, boost::none, epee::net_utils::ssl_support_t::e_ssl_support_disabled);
+
+    cryptonote::COMMAND_RPC_PING_STORAGE::request req = AUTO_VAL_INIT(req);
+    cryptonote::COMMAND_RPC_PING_STORAGE::response resp = AUTO_VAL_INIT(resp);
+
+    return net_utils::invoke_http_json_rpc("/json_rpc", "ping", req, resp, http_client, PING_TIMEOUT);
```

Comment:
Can this be simplified further by just using http_simple_client::connect()? We can get rid of the RPC boilerplate which is all noise in this instance.

Reply:
Hmm I see there's the equivalent GET response in the storage server now.

Reply:
I was thinking about that, but I wasn't sure if we wanted to use non-empty request/response or not.